app/api/logging/flask_logger.py

```diff
@@ -0,0 +1,108 @@
+"""Module for adding standard logging functionality to a Flask app.
+
+This module configures an application's logger to add extra data
+to all log messages. Flask application context data such as the
+app name and request context data such as the request method and
+request url rule are added to the log record.
+
+This module also configures the Flask application to log every
+non-404 request.
+
+Usage:
+    import api.logging.flask_logger as flask_logger
+
+    logger = logging.getLogger(__name__)
+    app = create_app()
+
+    flask_logger.init_app(logger, app)
+"""
+
+import logging
+
+import flask
+
+
+def init_app(app_logger: logging.Logger, app: flask.Flask) -> None:
+    """Initialize the Flask app logger.
+
+    Adds Flask app context data and Flask request context data
+    to every log record using log filters.
+    See https://docs.python.org/3/howto/logging-cookbook.html#using-filters-to-impart-contextual-information
+
+    Also configures the app to log every non-404 request using the given logger.
+
+    Usage:
+        import api.logging.flask_logger as flask_logger
+
+        logger = logging.getLogger(__name__)
+        app = create_app()
+
+        flask_logger.init_app(logger, app)
+    """
+    # Need to add filters to each of the handlers rather than to the logger itself, since
+    # messages are passed directly to the ancestor loggers’ handlers bypassing any filters
+    # set on the ancestors.
+    # See https://docs.python.org/3/library/logging.html#logging.Logger.propagate
+    for handler in app_logger.handlers:
+        handler.addFilter(_add_app_context_info_to_log_record)
+        handler.addFilter(_add_request_context_info_to_log_record)
+
+    # Use the app_logger to log every non-404 request before each request
+    # See https://flask.palletsprojects.com/en/2.2.x/api/#flask.Flask.before_request
+    app.before_request(lambda: _log_route(app_logger))
```

Comment:
Do we want to log at the end of the request too? Where we can log things like response code, size or response, duration.

Reply:
Yeah I think that's a good idea, but since that's new functionality (we weren't doing that before), I'll create a separate ticket for that to keep this PR as a refactor PR: https://github.com/navapbc/template-application-flask/issues/99